src/jshint.js

```diff
@@ -5227,7 +5244,7 @@ var JSHINT = (function() {
     if (!state.option.asi)
       nolinebreak(this);
 
-    if (state.tokens.next.id !== ";" && !state.tokens.next.reach &&
+    if (state.tokens.next.identifier &&
```

Comment:
Does this change to ASI handling work properly if the next line begins with a pre-increment or pre-decrement?

Reply:
@caitp I'm trying to think of an example, but maybe I'm misunderstanding you?

```js
var x = 0;
a: {
  break a
  --x
}
```

That doesn't seem right

Reply:
I couldn't see that this was for stmt("break") from mobile, I guess it makes more sense now that I can see it